sql-plugin/src/main/scala/com/nvidia/spark/rapids/GpuParquetScan.scala

```diff
@@ -1234,12 +1236,18 @@ case class GpuParquetMultiFilePartitionReaderFactory(
       val tc = TaskContext.get()
       val threadPool = MultiFileReaderThreadPool.getOrCreateThreadPool(numThreads)
       files.grouped(numFilesFilterParallel).map { fileGroup =>
+        // we need to copy the Hadoop Configuration because filter push down can mutate it,
+        // which can affect other threads.
         threadPool.submit(
-          new CoalescingFilterRunner(footerReadType, tc, fileGroup, conf, filters, readDataSchema))
+          new CoalescingFilterRunner(footerReadType, tc, fileGroup, new Configuration(conf),
+            filters, readDataSchema))
       }.toArray.flatMap(_.get())
     } else {
+      // we need to copy the Hadoop Configuration because filter push down can mutate it, which can
+      // affect other tasks which use it.
+      val hadoopConf = new Configuration(conf)
```

Comment:
This conf will get mutated for each file in `files` during the `.map` below, but I think this is OK since it's single-threaded in how that map is processed, and the way it is mutated shouldn't affect subsequent file processing.  Might want to add a comment explaining this is intentional.

Reply:
good point, updated.